baselines/ppo2/layers.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import tensorflow as tf
+
+from baselines.a2c.utils import ortho_init, fc
+from baselines.common.models import register
+
+
+@register("ppo_lstm")
+def ppo_lstm(nlstm=128, layer_norm=False):
```

Comment:
why are these new networks called `ppo_*`? They don't seem to be ppo-specific, I think they should be just as easily used with a2c, for instance?

Reply:
(and be moved into baselines.common.models?)

Reply:
oh I think I understand now, you are using different rnn functions and different way of managing state - only implemeneted in ppo - that's why these are ppo-specific for now. I have not read the rest of the PR yet, so more comments coming, but either way I'd like to have a singe view onto the RNN's in baselines (so if we are changing ppo, let's also change the rest of the algorithms compatible with rnns)

Reply:
I'm not sure here, but probably I can use RNN networks in `common.models` without additional implementations.

Reply:
@pzhokhov 
After I worked on it, I could use `baselines.a2c.utils.lstm` and `baselines.a2c.utils.lnlstm`. but it is unable to use RNNs in `baselines.common.models` because those declare `tf.placeholder`s.  

It is may arguable that using original `baselines.common.models` implementation and managing the memory of RNNs by another code. I had already implemented 'shared' version of it at in-house. It is complex enough. Because I had to manage not only initial_state, current_state, next_state of multiple RNNs but also name scope of copied network in this case.
(`policies.py` of the inhouse code: https://gist.github.com/gyunt/fd3ce10f2443a1ccc6d8253945904576  )